tests/Controller/PromotionCouponApiTest.php

```diff
@@ -0,0 +1,111 @@
+    ];
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_promotion_coupons_list_when_access_is_denied()
+    {
+        $promotions = $this->loadFixturesFromFile('resources/promotions.yml');
+        $promotion = $promotions['promotion2'];
+
+        $this->loadFixturesFromFile('resources/promotion_coupons.yml');
+
+        $this->client->request('GET', $this->getPromotionCouponUrl($promotion));
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_promotion_coupons()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $promotions = $this->loadFixturesFromFile('resources/promotions.yml');
+        $promotion = $promotions['promotion2'];
+        $this->loadFixturesFromFile('resources/promotion_coupons.yml');
+
+        $this->client->request('GET', $this->getPromotionCouponUrl($promotion), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'promotion_coupon/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_promotion_when_it_does_not_exist()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $promotions = $this->loadFixturesFromFile('resources/promotions.yml');
+        $promotion = $promotions['promotion2'];
+
+        $this->client->request('GET', sprintf('/api/v1/promotions/%s/coupons/-1', $promotion->getCode()), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_promotion_coupon()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $promotions = $this->loadFixturesFromFile('resources/promotions.yml');
+        $promotion = $promotions['promotion2'];
+
+        $promotion_coupons = $this->loadFixturesFromFile('resources/promotion_coupons.yml');
+        $coupon = $promotion_coupons['promotionCoupon1'];
+
+        $this->client->request('GET', $this->getPromotionCouponUrl($promotion, $coupon), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'promotion_coupon/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @param PromotionInterface $promotion
+     * @param PromotionCouponInterface|null $coupon
+     *
+     * @return string
+     */
+    private function getPromotionCouponUrl(PromotionInterface $promotion, PromotionCouponInterface $coupon = null)
+    {
+        if (null === $coupon) {
```

Comment:
Instead of if condition it would be better to have two functions, where one will use another